File: 01-week-1--starter-algorithms/04-day-5--find-shortest-string/javascript/find_shortest_string.js
function findShortestString(arr) {
  // type your code here
  // let shortestString is the first item from arr
  // loop the array and compare the length of each item with the length of shortestString
  //  if arr[i].length < shortestString.length, shortestString = arr[i]
  // return shortestString

  let shortestString = arr[0];

  for(let i = 1; i < arr.length; i++){
    if(arr[i].length < shortestString.length) shortestString = arr[i];
  }

  return shortestString;


}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'a'");
  console.log("=>", findShortestString(['aaa', 'a', 'bb', 'ccc']));

  console.log("");

  console.log("Expecting: 'hi'");
  console.log("=>", findShortestString(['cat', 'hi', 'dog', 'an']));

  console.log("");

  console.log("Expecting: 'lily'");
  console.log("=>", findShortestString(['flower', 'juniper', 'lily', 'dadelion']));

  // BENCHMARK HERE
}

module.exports = findShortestString;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 01-week-1--starter-algorithms/04-day-5--find-shortest-string/ruby/find_shortest_string.rb
def find_shortest_string(arr)
  # type your code in here

  shortestStr = arr[0]

  for i in 1..arr.length-1 do
    shortestStr = arr[i] if arr[i].length < shortestStr.length
  end

  shortestStr
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'a'"
  puts "=>", find_shortest_string(['aaa', 'a', 'bb', 'ccc'])

  puts

  puts "Expecting: 'hi'"
  puts "=>", find_shortest_string(['cat', 'hi', 'dog', 'an'])

  puts

  puts "Expecting: 'lily'"
  puts "=>", find_shortest_string(['flower', 'juniper', 'lily', 'dadelion'])

  # Don't forget to add your own!

  # BENCHMARK HERE
end

# Please add your pseudocode to this file
# And a written explanation of your solution
